internal/bminventory/inventory.go

```diff
@@ -3135,3 +3124,67 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func (b bareMetalInventory) PermanentlyDeleteUnregisteredClustersAndHosts() {
```

Comment:
why it is part of bm-inventory? 
i think better to move to cluster monitor for example

Reply:
let monitor to handle it

Reply:
sounds like inventory is not the best place for it for sure. move it to cluster maybe

Reply:
+1

Reply:
Cluster monitor runs every 7 seconds, we don't need this query to be executed each 7 seconds. If you are talking about combining between the two queries I think it will be too much. Especially because this query also does not filter by states.

Reply:
So let's create a new monitor with all this logic, bminventory is too bloated already